parcels/particlesets/particlesetsoa.py

```diff
@@ -0,0 +1,528 @@
+                kwargs[v] = vars[v]
+
+        if restart:
+            pclass.setLastID(0)  # reset to zero offset
+        else:
+            vars['id'] = None
+
+        return cls(fieldset=fieldset, pclass=pclass, lon=vars['lon'], lat=vars['lat'],
+                   depth=vars['depth'], time=vars['time'], pid_orig=vars['id'],
+                   lonlatdepth_dtype=lonlatdepth_dtype, repeatdt=repeatdt, **kwargs)
+
+    def to_dict(self, pfile, time, deleted_only=False):
+        """
+        Convert all Particle data from one time step to a python dictionary.
+        :param pfile: ParticleFile object requesting the conversion
+        :param time: Time at which to write ParticleSet
+        :param deleted_only: Flag to write only the deleted Particles
+        returns two dictionaries: one for all variables to be written each outputdt,
+         and one for all variables to be written once
+        """
+        return self._collection.toDictionary(pfile=pfile, time=time,
+                                             deleted_only=deleted_only)
+
+    @property
+    def size(self):
+        # ==== to change at some point - len and size are different things ==== #
+        return len(self._collection)
+
+    def __repr__(self):
+        return "\n".join([str(p) for p in self])
+
+    def __len__(self):
+        return len(self._collection)
+
+    def __sizeof__(self):
+        return sys.getsizeof(self._collection)
+
+    def __iadd__(self, particles):
+        """Add particles to the ParticleSet. Note that this is an
+        incremental add, the particles will be added to the ParticleSet
+        on which this function is called.
+
+        :param particles: Another ParticleSet containing particles to add
+                          to this one.
+        :return: The current ParticleSet
+        """
+        self.add(particles)
+        return self
+
+    def add(self, particles):
+        """Add particles to the ParticleSet. Note that this is an
+        incremental add, the particles will be added to the ParticleSet
+        on which this function is called.
+
+        :param particles: Another ParticleSet containing particles to add
+                          to this one.
+        :return: The current ParticleSet
+        """
+        if isinstance(particles, BaseParticleSet):
+            particles = particles.collection
+        self._collection += particles
+        return self
+
+    def remove_indices(self, indices):
+        """Method to remove particles from the ParticleSet, based on their `indices`"""
+        if type(indices) in [int, np.int32, np.intp]:
+            self._collection.remove_single_by_index(indices)
+        else:
+            self._collection.remove_multi_by_indices(indices)
+
+    def remove_booleanvector(self, indices):
+        """Method to remove particles from the ParticleSet, based on an array of booleans"""
+        self.remove_indices(np.where(indices)[0])
```

Comment:
I removed the code comments, but will restate what they said here: the functions need to be replaced / removed in the future.